sbatson5/firestore-jest-mock
# Migrate to exports

# Description

This PR attempts to support both `import` and `require` syntax, following [this blog post](https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html) as a guide.
This also updates a lot of dependencies that have gone out of date and should prepare use for migrating to firebase 9 and Typescript (although that will be a breaking/major change/release).

This also drops support for Node 10, 11 and 12, and adds support for 16, 17 and 18

## Related issues

Fixes https://github.com/Upstatement/firestore-jest-mock/issues/144

## How to test

Can someone point their repo to this branch and make sure it is backwards compatible with `require` and then works with `import` syntax?

## TODOs
